test/e2e/e2e_test.go

```diff
@@ -91,8 +98,138 @@ func TestCatalog(t *testing.T) {
 	assert.NotEmpty(t, plans.Items, "Expected service plans to exist")
 }
 
-// setupTest will create a new namespace for a single test and deploy the
-// broker inside.
+func TestProvision(t *testing.T) {
+	t.Parallel()
+
+	namespace := setupTest(t)
+	defer cleanupTest(t)
+
+	_, err := setupServiceInstance(t, namespace)
+	assert.NoError(t, err, "Expected instance to have been provisioned")
+}
+
+func TestUpdate(t *testing.T) {
+	t.Parallel()
+
+	namespace := setupTest(t)
+	defer cleanupTest(t)
+
+	// Create service instance with plan size M10
+	serviceinstance, err := setupServiceInstance(t, namespace)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Ensure cluster is in the correct state
+	cluster, err := svcatClient.ServicecatalogV1beta1().ServiceInstances(namespace).Get(serviceinstance.GetName(), metav1.GetOptions{})
+	if !assert.NoError(t, err) {
+		return
+	}
+	assert.Equal(t, "M10", cluster.Spec.ClusterServicePlanExternalName)
+
+	// Make changes in the service instance
+	serviceinstance.Spec.ClusterServicePlanExternalName = "M20"
+
+	// Update the service instance with this change
+	updatedServiceInstance, err := svcatClient.ServicecatalogV1beta1().ServiceInstances(namespace).Update(serviceinstance)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Wait up to 25 minutes for the service instance to have been updated.
+	err = testutil.Poll(25, func() (bool, error) {
```

Comment:
[q] I see that the callback inside returns 'true' on the first invocation - why do we need to call `Poll` then?